pkg/gpt/gpt.go

```diff
@@ -261,34 +308,43 @@ func Write(w io.WriterAt, g *GPT) error {
 	return err
 }
 
-// New reads in the primary and backup GPT from a disk and returns a pointer to them.
+// New reads in the MBR, primary and backup GPT from a disk and returns a pointer to them.
 // There are some checks it can apply. It can return with a
 // one or more headers AND an error. Sorry. Experience with some real USB sticks
 // is showing that we need to return data even if there are some things wrong.
-func New(r io.ReaderAt) (*GPT, *GPT, error) {
+func New(r io.ReaderAt) (*PartitionTable, error) {
+	var p = &PartitionTable{}
+	var mbr = &MBR{}
+	n, err := r.ReadAt(mbr[:], 0)
+	if n != BlockSize || err != nil {
+		return p, err
+	}
+	p.MasterBootRecord = mbr
```

Comment:
Do I understand correctly that this just reads in and writes out the same MBR?

Reply:
That's what I understood as well.

Reply:
yes, potentially, although, note, you can modify later if you want.